### PR TITLE
Entity: Replaced a mutexed counter with a std::atomic.

### PR DESCRIPTION
**Pull request by madmaxoft**
Fixes a warning from clang.

Also prettified one ugly one-line macro.

### CODE REVIEW

File: src/Entities/Entity.cpp [hunk 1]
```diff
@@ -19,15 +19,21 @@
...
 
 
 
 
+////////////////////////////////////////////////////////////////////////////////
+// cEntity:
+
 cEntity::cEntity(eEntityType a_EntityType, double a_X, double a_Y, double a_Z, double a_Width, double a_Height):
-	m_UniqueID(INVALID_ID),  // Proper ID will be assigned later in the constructor code
+	m_UniqueID(GetNextUniqueID()),
```
Comment (peterbell10) [inline src/Entities/Entity.cpp]: Out of curiosity, why not just `++m_EntityCount`?

Comment (NiLSPACE) [inline src/Entities/Entity.cpp]: It's not thread-safe. That's why there was a lock previously.

Comment (peterbell10) [inline src/Entities/Entity.cpp]: To be clear, I meant making `m_EntityCount` atomic and using the atomic pre-increment.

Comment (madmaxoft) [inline src/Entities/Entity.cpp]: Because the `m_EntityCount` would still be a de-facto global object requiring a global constructor.

Comment (peterbell10) [inline src/Entities/Entity.cpp]: Ah, makes sense. Thank you.

**Pull request merged**